src/coreclr/jit/gentree.cpp

```diff
@@ -6507,9 +6507,16 @@ GenTree* Compiler::gtNewLclvNode(unsigned lnum, var_types type DEBUGARG(IL_OFFSE
         // should be able to remove this exception and handle the assignment mismatch in
         // Lowering.
         LclVarDsc* varDsc = lvaGetDesc(lnum);
+#if FEATURE_SIMD
+        // We can additionally have a SIMD12 that was widened to a SIMD16, generally as part of lowering
+        assert((type == varDsc->lvType) || ((type == TYP_SIMD16) && (varDsc->lvType == TYP_SIMD12)) ||
```

Comment:
Because we weren't introducing `TYP_SIMD12` nodes in lowering, it was all happening either earlier (in which case the nodes naturally lowered as part of normal traversal) or being handled in codegen (leading to less efficient codegen).

Reply:
Fixed the assertion to remove duplication.